sklearn/compose/_column_transformer.py

```diff
@@ -265,6 +265,10 @@ def set_output(self, transform=None):
         transform : {"default", "pandas"}, default=None
             Configure output of `transform` and `fit_transform`.
 
+            - `"default"`: Output of an un-configured transformer
+            - `"pandas"`: DataFrames output
+            - `"None"`: No-op
```

Comment:
After reading the "no op" explanation I was left wondering how "no-op" was different from "default". Looking at the code I think the answer is "they are the same". If that is correct, can we say "`None`: Default output (format) of a transformer"?

I think I like "Default output format of a transformer" better than "Output of an un-configured transformer", but I am not sure if they are equivalent? As in, I could imagine that there is a subtle difference between "unconfigured" and "default"?,

Reply:
I think `no-op` is supposed to mean `transform=None` would do nothing, but `transform="default"` sets the output to be `pandas`.

Reply:
Ouf, that is subtle indeed. Why does `None` exist as an option if it does nothing (not even reset the state to "unconfigured")?


> but `transform="default"` sets the output to be `pandas`

is this right? I was expecting `"default"` to be "not pandas"/numpy because there is an option to set pandas

Reply:
> is this right? 

Sorry, typo, but you got the intention :D 

> Why does None exist as an option if it does nothing

This method might start having other args as well, and then we'd want to be able to set only one thing, while leaving the rest unchanged:

```py
def f(a=None, b=None):
    ...

# This would not change the value set by the method for `b`
f(a="value")
```

Reply:
> Why does None exist as an option if it does nothing (not even reset the state to "unconfigured")?

In the future, we may want to add `predict` to `set_output`:

```python
pipe.set_output(transform="pandas")
pipe.set_output(predict="pandas")
```

The second call with `predict="pandas"` should not adjust the configuration for `transform`.

> is this right? I was expecting "default" to be "not pandas"/numpy because there is an option to set pandas

I took into account existing third party estimators that only return DataFrames as the default. If we had "default" == "numpy", then they would not be following the `set_output` API. I did not want to add more work for third party libraries to convert their DataFrames into NumPy arrays just to conform to the API.

Secondly, even in scikit-learn, we have three possible return containers for `transform="default"`:

- With the recent Array API support: Any Array API container such as `cupy.array_api.Array`
- NumPy's ndarray
- SciPy's sparse matrix

Reply:
From this discussion, I updated the docstring to:

```bash
- `"default"`: Default output format of a transformer
- `"pandas"`: DataFrame output
- `None`: Transform configuration is unchanged